Src/InGreed.Api/Controllers/OpinionController.cs

```diff
@@ -0,0 +1,70 @@
+using InGreed.Api.Contracts.Opinion;
+using InGreed.Domain.Models;
+using InGreed.Logic.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using InGreed.Logic.Enums.Opinion;
+
+
+namespace InGreed.Api.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class OpinionController : ControllerBase
+{
+    IOpinionService _opinionService;
+
+    public OpinionController(IOpinionService opinionService)
+    {
+        _opinionService = opinionService;
+    }
+
+    [HttpGet("{id}")]
+    public IActionResult GetById(int id)
+    {
+        Opinion? result = _opinionService.GetById(id);
+        if (result is null) return BadRequest();

```

Comment:
I would go with NotFound() here but it really doesn't matter.

Reply:
Ok, will change in next commit.